espnet2/tasks/hubert.py

```diff
@@ -282,6 +282,9 @@ def build_collate_fn(
             pad=args.collate_fn_conf.get("pad", False),
             rand_crop=args.collate_fn_conf.get("rand_crop", True),
             crop_audio=not args.collect_stats,
+            kernel_size=args.collate_fn_conf.get("kernel_size", 25),
```

Comment:
Can you calculate the `kernel_size / stride` from the model config? Is it possible to get `sample_rate` from somewhere?
I'm concerned that manually specifying it from the configuration make it difficult. If someone is not familiar, it is very easy to forget to define them in config.

Reply:
Possible, one option is when I build the model, I directly update the options in collate_fn_conf of argparse.Namespace. Then when create collate_fn, we reuse the updated one

Reply:
It is possible as the collate function are created after the model initialization

Reply:
Do you think this would be good option? I'm still unsure of any potential issue with that

Reply:
According to the order in `abs_task.py`, collate_fn is built after the model initialization. IMO, either modifying the `args.collate_fn_conf` or adding another argument for `build_collate_fn()` is fine

Reply:
Sounds good, I will do with the first option then as the second one kinds of break the code logic a bit in the main function.